src/confirm.rs

```diff
@@ -0,0 +1,116 @@
+use std::time::Duration;
+use futures::{Future, Stream};
+use api::{Eth, EthFilter, Namespace};
+use types::{H256, U256, TransactionRequest};
+use {Transport, Error};
+
+pub fn wait_for_confirmations<'a, T, F, V>(transport: T, poll_interval: Duration, confirmations: usize, check: V)
+  -> Box<Future<Item = (), Error = Error> + 'a> where
+  T: 'a + Transport + Clone,
+  F: 'a + Future<Item = Option<U256>, Error = Error>,
```

Comment:
`IntoFuture` is more general, then you can just return `Result` from that function